crates/astria-sequencer-relayer/src/relayer/mod.rs

```diff
@@ -273,13 +296,63 @@ impl Relayer {
         }
         Ok(())
     }
+
+    async fn determine_last_submitted_sequencer_height(
+        &self,
+    ) -> eyre::Result<Option<SequencerHeight>> {
+        let presubmit_path = self.presubmit_path.clone();
+        let presubmit_state = crate::utils::flatten(
+            tokio::task::spawn_blocking(move || read_state_file(presubmit_path)).await,
+        )
+        .wrap_err("failed reading presubmit state from provided file")?;
+        let postsubmit_path = self.postsubmit_path.clone();
+        let postsubmit_state = crate::utils::flatten(
+            tokio::task::spawn_blocking(move || read_state_file(postsubmit_path)).await,
+        )
+        .wrap_err("failed reading postsubmit state from provided file")?;
+
+        match (presubmit_state, postsubmit_state) {
+            (None, None) => Ok(None),
+            (Some(pre), Some(post)) if write::are_states_consistent(pre, post) => {
+                let determined_height = post.sequencer_height();
+                let height = determined_height.try_into().wrap_err_with(|| {
+                    format!(
+                        "last submitted sequencer height was determined to be \
+                         `{determined_height}`, but couldn't be turned into cometbft/sequencer \
+                         height"
+                    )
+                })?;
+                Ok(Some(height))
+            }
+            (..) => bail!(
+                "manual intervention required: pre- and post-submit states must either both be \
+                 unset (their files present but empty),
+                    or both be set and consistent (files present and deserializable, equal \
+                 sequencer heights, celestia heights pre-submission less
+                    then post-submission)"
```

Comment:
the files have to exist before relayer is started for the first time? or does relayer init the files?